lib/friends/introvert.rb

```diff
@@ -777,5 +779,52 @@ def thing_with_name_in(type, text)
     def bad_line(expected, line_num)
       raise FriendsError, "Expected \"#{expected}\" on line #{line_num}"
     end
+
+    def decide_default_location_output(activity)
+      @activities = stable_sort(@activities)
+      existing_activities = @activities - [activity]
+
+      str = "Default location"
+      if activity != @activities.first
+        str += " from #{earliest_consecutive_default_location_activity_date(activity, existing_activities)} to"
+        str += " #{next_activity_date_with_different_default_location(activity, existing_activities) || 'present'}"
+      end
+
+      str += " already" if last_default_location_same_as_added_default_location?(activity, existing_activities)
+      str += " set to: \"#{activity.default_location}\""
+
+      @output << str
+    end
+
+    def last_default_location_same_as_added_default_location?(activity, existing_activities)
+      last_default_location_activity = existing_activities.select { |a| a.default_location && (a.date < activity.date) }.first
+      if last_default_location_activity
+        last_default_location_activity.default_location == activity.default_location
+      end
+    end
+
+    def earliest_consecutive_default_location_activity_date(activity, existing_activities)
+      # binding.pry
+      earlier_default_location_activities = existing_activities.select { |a| a.default_location && (a.date < activity.date) }
+      consecutively_earliest_activity_with_same_default_location = nil
+      earlier_default_location_activities.each do |act|
+        break if act.default_location != activity.default_location
+        consecutively_earliest_activity_with_same_default_location = act if act.default_location == activity.default_location
+      end
+      if consecutively_earliest_activity_with_same_default_location
+        return consecutively_earliest_activity_with_same_default_location.date
+      else
+        return activity.date
+      end
+    end
+
+    def next_activity_date_with_different_default_location(activity, existing_activities)
+      next_activity_with_different_default_location = existing_activities.select { |a| a.default_location && (a.date > activity.date) && a.default_location != activity.default_location }.first
+      if next_activity_with_different_default_location
```

Comment:
A failed conditional evaluates to `nil`, so this can just be:

```ruby
if next_activity_with_different_default_location
  next_activity_with_different_default_location.date
end
```

or

```ruby
next_activity_with_different_default_location&.date
```

Reply:
Gotcha - I don't use the `&.` enough. Making a mental note to use it more :)